smt.go

```diff
@@ -463,10 +475,14 @@ func (smt *SMT) ProveClosest(path []byte) (
 	}
 
 	// Retrieve the closest path and value hash if found
-	if node == nil {
-		panic("no leaf node found")
+	if node == nil { // tree was empty
```

Comment:
I think we should print that the tree was empty.

If/when we hit bugs, this is the sort of thing that helps find it.

Reply:
I personally would be against including print statements in the library itself only included when doing debugging otherwise they would just bloat out the SMT during normal use. I think having added the new return specific to empty trees allows us to determine the outcome without needing the print statement:
1. Was the `closestPath` returned a placeholder & is the root hash a placeholder & was `closestValueHash` returned nil?
  - Tree 100% empty
2. Otherwise tree is non-empty